src/components/ToDoItem.js

```diff
@@ -3,29 +3,120 @@ import './ToDoItem.css'
 
 
 
-const ToDoItem = (props) => {
-	return (
-		<div className="todo-item">
-			<input 
-				type="checkbox" 
-				checked={props.checked}
-				onChange={() => props.onCheck(props.date)}
+class ToDoItem extends React.Component {
+
+	state = {
+		edit: false,
+		saveChanges: false,
+		value: '',
+		editedValue: ''
+	}
+
+	handleEdit = () => {
+		this.setState({ 
+			edit: !this.state.edit 
```

Comment:
you must use functional setState in such cases (when new state is a transformed result of previous state)